lib/extensions/ar_virtual.rb

```diff
@@ -386,6 +386,14 @@ def virtual_attribute_names
       end
     end
 
+    def attributes_builder
+      @attributes_builder ||= ::ActiveRecord::AttributeSet::Builder.new(attribute_types, primary_key) do |name|
+        unless columns_hash.key?(name) || virtual_attribute?(name)
```

Comment:
I would normally whine about unless with an `||` but since the original rails code has unless... I'm good with keeping it as close to the original code.  I'll just say my brain still needs to convert this to an if.